.github/workflows/ci.yml

```diff
@@ -443,8 +598,68 @@ jobs:
         rm -rf $(find . -maxdepth 1 ! -name "build" ! -name "llvm" ! -name "clang" ! -name ".")
         cd ../
 
+    - name: Build LLVM/Cling on Unix systems if the cache is invalid (emscripten)
+      if: ${{ runner.os != 'windows' && steps.cacheemscripten.outputs.cache-hit != 'true' && matrix.emscripten == 'On' }}
+      run: |
+        cling_on=$(echo "${{ matrix.cling }}" | tr '[:lower:]' '[:upper:]')
+        if [[ "${cling_on}" == "ON" ]]; then
+          git clone https://github.com/root-project/cling.git
+          cd ./cling 
+          git checkout tags/v${{ matrix.cling-version }}
+          cd ..
+          git clone --depth=1 -b cling-llvm${{ matrix.clang-runtime }} https://github.com/root-project/llvm-project.git
+        else # repl
+          git clone --depth=1 -b release/${{ matrix.clang-runtime }}.x https://github.com/llvm/llvm-project.git
+        fi
+        cd llvm-project
+        # Build
+        mkdir build
+        if [[ "${cling_on}" == "ON" ]]; then
+          cd build
+          #FIXME: This line shouldn't be this line, but emcmake wouldn't recognise new line \
+          cmake ../llvm -DLLVM_EXTERNAL_PROJECTS=cling \
+          -DLLVM_EXTERNAL_CLING_SOURCE_DIR=../../cling \
+          -DCMAKE_BUILD_TYPE=Release  \
+          -DLLVM_HOST_TRIPLE=wasm32-unknown-emscripten \
+          -DLLVM_TARGETS_TO_BUILD="${{ matrix.llvm_targets_to_build }}" \
+          -DLLVM_ENABLE_LIBEDIT=OFF \
+          -DLLVM_ENABLE_PROJECTS="${{ matrix.llvm_enable_projects }}" \
+          -DLLVM_ENABLE_ZSTD=OFF \
+          -DLLVM_ENABLE_LIBXML2=OFF \
+          -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+          -DCLANG_ENABLE_ARCMT=OFF \
+          -DCLANG_ENABLE_BOOTSTRAP=OFF 
+          cmake --build . --target clang --parallel ${{ env.ncpus }}
+          cmake --build . --target cling --parallel ${{ env.ncpus }}
+          # Now build gtest.a and gtest_main for CppInterOp to run its tests.
+          cmake --build . --target gtest_main --parallel 2
```

Comment:
Currently we don't provide a cling based emscripten build for cppinterop. Can we not error out here and maybe add a todo for whenever this is done.

Reply:
@anutosh491 This is putting it what is believed to be needed in the ci for when we do have a cling based build in the future. No cling based jobs will be added to this PR.